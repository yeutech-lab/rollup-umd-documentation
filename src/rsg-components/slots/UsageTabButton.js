import React from 'react'; // eslint-disable-line no-unused-vars
import PropTypes from 'prop-types';
import TabButton from '../TabButton'; // eslint-disable-line no-unused-vars

const UsageTabButton = (props) => {
  const component = props.props;
  const showButton = component.props || (component.methods && component.methods.length > 0);
  return showButton ? <TabButton {...props}>Props & methods</TabButton> : null;
};

UsageTabButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  props: PropTypes.shape({
    props: PropTypes.array,
    methods: PropTypes.array,
  }).isRequired,
  active: PropTypes.bool,
};

export default UsageTabButton;
