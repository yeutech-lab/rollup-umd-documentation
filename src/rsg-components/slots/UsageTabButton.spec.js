import React from 'react';
import UsageTabButton from './UsageTabButton';

const props = {
	name: 'Pizza',
	onClick: () => {},
};

it('should renderer a button', () => {
	const actual = shallow(<UsageTabButton {...props} props={{ props: [{ name: 'foo' }] }} />);

	expect(actual).toMatchSnapshot();
});

it('should renderer null if there are not props or methods', () => {
	const actual = shallow(<UsageTabButton {...props} props={{}} />);

	expect(actual.getElement()).toBeFalsy();
});
