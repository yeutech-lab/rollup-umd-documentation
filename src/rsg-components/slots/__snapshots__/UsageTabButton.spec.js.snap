// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renderer a button 1`] = `
<Styled(TabButton)
  name="Pizza"
  onClick={[Function]}
  props={
    Object {
      "props": Array [
        Object {
          "name": "foo",
        },
      ],
    }
  }
>
  Props & methods
</Styled(TabButton)>
`;
